examples/common/python/work_order_receipt/work_order_receipt_request.py

```diff
@@ -0,0 +1,130 @@
+        requester_nonce = nonce
+        if nonce is None:
+            requester_nonce = str(random.randint(1, 10**10))
+        public_key =  signing_key.GetPublicKey().Serialize()
+        wo_receipt_reqest = {
+            "workOrderId": wo_request_params["workOrderId"],
+            # TODO: workerServiceId is same as worker id,
+            # needs to be updated with actual service id
+            "workerServiceId": worker_id,
+            "workerId": worker_id,
+            "requesterId": wo_request_params["requesterId"],
+            "receiptCreateStatus": receipt_create_status,
+            "workOrderRequestHash": final_hash_str,
+            "requesterGeneratedNonce": requester_nonce,
+            "signatureRules": self.HASHING_ALGORITHM + "/" + self.SIGNING_ALGORITHM,
+            "receiptVerificationKey": public_key
+        }
+        wo_receipt_str = wo_receipt_reqest["workOrderId"] + \
+            wo_receipt_reqest["workerServiceId"] + \
+            wo_receipt_reqest["workerId"] + \
+            wo_receipt_reqest["requesterId"] + \
+            str(wo_receipt_reqest["receiptCreateStatus"]) + \
+            wo_receipt_reqest["workOrderRequestHash"] + \
+            wo_receipt_reqest["requesterGeneratedNonce"]
+        wo_receipt_bytes = bytes(wo_receipt_str, "UTF-8")
+        wo_receipt_hash = crypto.compute_message_hash(wo_receipt_bytes)
+        status, wo_receipt_sign = self.sig_obj.generate_signature(
+            wo_receipt_hash,
+            signing_key
+        )
+        if status == False:
+            wo_receipt_reqest["requesterSignature"] = ""
+        else:
+            wo_receipt_reqest["requesterSignature"] = wo_receipt_sign
+        return wo_receipt_reqest
+
+    def update_receipt(self, wo_resp, signing_key):
+        """
+        Update the existing work order receipt with the status as in wo_resp
+        Parameters:
+            - wo_resp is dictionary containing json rpc response
+            of the work order execution as defined in EEA sepc 6.1.2
+            - signing_key is private key
+        Returns jrpc request of type dictionary
+        """
+        wo_resp_str = json.dumps(wo_resp)
+        wo_resp_bytes = bytes(wo_resp_str, "UTF-8")
+        wo_resp_hash = crypto.compute_message_hash(wo_resp_bytes)
+        wo_resp_hash_str = crypto.byte_array_to_hex(wo_resp_hash)
+        receipt_status = None
+
+        public_key = signing_key.GetPublicKey().Serialize()
+        wo_id = None
+        if "error" in wo_resp and \
+            wo_resp["error"]["code"] != WorkOrderStatus.PENDING.value:
+            receipt_status = ReceiptCreateStatus.FAILED.value
+            wo_id = wo_resp["error"]["data"]
+        else:
+            receipt_status = ReceiptCreateStatus.PROCESSED.value
+            wo_id = wo_resp["result"]["workOrderId"]
+        
+        wo_receipt_update = {
+            "workOrderId": wo_id,
+            "updaterId": wo_id,
+            "updateType": receipt_status,
```

Comment:
Spec is defined with 0 to 5 defined values and 5 to 255 are reserved and 255 for any states and above 255 are application specific. Since  our reference implementation EM is updating it knows about 0 to 4.